packages/gui/src/components/chiatools/logViewer/LogViewerToolbar.tsx

```diff
@@ -0,0 +1,170 @@
+import { Flex } from '@chia-network/core';
+import { Trans } from '@lingui/macro';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import FileDownloadIcon from '@mui/icons-material/FileDownload';
+import RefreshIcon from '@mui/icons-material/Refresh';
+import {
+  Box,
+  TextField,
+  IconButton,
+  Tooltip,
+  FormControl,
+  Select,
+  MenuItem,
+  Checkbox,
+  ListItemText,
+  Pagination,
+  Paper,
+} from '@mui/material';
+import debounce from 'lodash/debounce';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { LogLevel, LogViewerFilter, PaginationInfo } from './LogViewerTypes';
+
+type LogViewerToolbarProps = {
+  filter: LogViewerFilter;
+  onFilterChange: (filter: LogViewerFilter) => void;
+  onRefresh: () => void;
+  onExport: () => void;
+  onCopy: () => void;
+  loading?: boolean;
+  pagination: PaginationInfo;
+  onPageChange: (page: number) => void;
+};
+
+export default function LogViewerToolbar({
+  filter,
+  onFilterChange,
+  onRefresh,
+  onExport,
+  onCopy,
+  loading,
+  pagination,
+  onPageChange,
+}: LogViewerToolbarProps) {
+  const [searchText, setSearchText] = useState('');
+
+  // Create a ref to store the debounced function
+  const debouncedRef = React.useRef<any>();
+
+  // Create debounced filter update
+  const debouncedFilterUpdate = useCallback(
+    (inputText: string) => {
+      if (debouncedRef.current) {
+        debouncedRef.current.cancel();
+      }
+
+      debouncedRef.current = debounce((filterText: string) => {
+        onFilterChange({
+          ...filter,
+          searchText: filterText,
+        });
+      }, 300);
+
+      debouncedRef.current(inputText);
+    },
+    [filter, onFilterChange],
+  );
```

Comment:
Generating a new debounce function and cancelling old one everytime user inputs is a bit inefficient.

I suggest the following code:
```typescript
const debouncedFilterFunc = useMemo(() => {
  return debounce((filterText: string) => {
    onFilterChange({
      ...filter,
      searchText: filterText,
    });
  }, 300);
}, [filter, onFilterChange]);
...
useEffect(() => () => {
  debouncedFilterFunc.cancel();
}, [debouncedFilterFunc]);
```

Reply:
refactored the debounce to use this code